src/types/chat.ts

```diff
@@ -17,7 +20,25 @@ export type PrivateChatMessage = BaseChatMessage & {
 
 export type VenueChatMessage = BaseChatMessage;
 
-export type ChatMessage = PrivateChatMessage | VenueChatMessage;
+export type VoteInPoll = {
+  questionId: number;
+  pollId: string;
+};
+
+type Vote = {
+  questionId: number;
+  userId: string;
+};
+
+export type PollMessage = BaseChatMessage & {
+  poll: PollValues;
+  votes: Vote[];
+};
+
+export type ChatMessage = PrivateChatMessage | VenueChatMessage | PollMessage;
+
+export const isPollMessage = (r: unknown): r is PollMessage =>
+  typeof r === "object" && isTruthy(r) && r.hasOwnProperty("poll");
```

Comment:
> Though this still feels kind of hacky.. instead of adding a boolean to our message types, we should be more flexible/future thinking and just give them a specific `type`.
> 
> Then we could have similar code to above, but where we check `type`. This would make our message API simpler in general, and is cleaner/more robust.
> 
> For now I suggest we just add it to the 'question' messages here, and to the poll messages in #1343. It'll be way easier to implement now before they get shipped out than to have to deal with backfilling/handling legacy data across all of our environments in some future point once they hit production.
> 
> @mike-lvov Thoughts? I feel fairly strongly about this one.
> 
> _Originally posted by @0xdevalias in https://github.com/sparkletown/sparkle/pull/1424#discussion_r640241413_

Reply:
Done in https://github.com/sparkletown/sparkle/pull/1343/commits/3e8ec8791c5e715576d66645965ca8d7cb65a00a